tiledb/sm/c_api/tiledb.h

```diff
@@ -3201,6 +3234,468 @@ TILEDB_EXPORT int32_t tiledb_array_encryption_type(
+ * @param budget_val The result size budget (in bytes) for the values to
+ *     be retrieved for the input attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_result_budget_var(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    const char* attr_name,
+    uint64_t* budget_off,
+    uint64_t* budget_val);
+
+/**
+ * Gets the memory budget (in bytes) that dictates the maximum amount
+ * of bytes to be read and decompressed for the attributes during
+ * reads.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint64_t budget, budget_var;
+ * tiledb_subarray_partitioner_get_memory_budget(
+ *     ctx, partitioner, &budget, &budget_var);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The partitioner.
+ * @param budget The memory budget for a fixed-sized attribute or the
+ *     offsets of a var-sized attribute.
+ * @param budget_var The memory budget for a var-sized attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_memory_budget(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    uint64_t* budget,
+    uint64_t* budget_var);
+
+/**
+ * Sets the result budget (in bytes) for a given fixed-sized attribute.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_set_result_budget(
+ *     ctx, partitioner, "a", 10000000);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The partitioner.
+ * @param attr_name The name of the attribute to set the budget for.
+ * @param budget The result size budget (in bytes) for this attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_result_budget(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    const char* attr_name,
+    uint64_t budget);
+
+/**
+ * Sets the result budget (in bytes) for for a given var-sized attribute.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_set_result_budget_var(
+ *     ctx, partitioner, "a", 10000000, 10000000);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The subarray partitioner.
+ * @param attr_name The name of the attribute to set the budget for.
+ * @param budget_off The result size budget (in bytes) for the
+ *     offsets of this attribute.
+ * @param budget_val The result size budget (in bytes) for the
+ *     values of this attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_result_budget_var(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    const char* attr_name,
+    uint64_t budget_off,
+    uint64_t budget_val);
+
+/**
+ * Sets the memory budget (in bytes) that dictates the maximum amount
```

Comment:
Probably should add a brief comment here or elsewhere on how this relates to/interacts with the new memory budget config params.